src/big/big_float.cr

```diff
@@ -18,7 +18,39 @@ struct BigFloat < Float
   end
 
   def initialize(num : Number)
-    LibGMP.mpf_init_set_d(out @mpf, num.to_f64)
+    # XXX: this case is workaround of Crystal's unrealiable method overloading
+    # remove it when separate BigFloat.new(BigInt) will pass the spec
+    case num
+    when BigInt
+      LibGMP.mpf_init(out @mpf)
```

Comment:
@RX14  feel free to merge this PR firstly. 
There a lot of common code, because we have no assumptions on its pulling order.

I will rebase my PR anyway.